### PR TITLE
Clear step

### LINKED ISSUE #1369: Writing UI test and making UI more stable
With #1366 we were able to set up a basic browser-based UI test and it works flawlessly. So lets build more tests and make the UI stable.
Test yet to be built:

- [ ] Clicking add step adds a new box and a different image

- [ ] Clicking clear step button clear all the step

- [ ] Clicking Delete Step removes the image

- [ ] Select a module to have all the modules with the correct name(we faced issues when modules have which is undefined in the past ).

- [ ] Insert step works as expected.

- [ ] Input types(range,text etc) of all the modules are correct-
we need to fix this before we can write test on it

### PR DESCRIPTION
**Pull request by ataata107**
Concerns #1369  Clear Step

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [X] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm test`
* [X] code is in uniquely-named feature branch and has no merge conflicts
* [X] PR is descriptively titled
* [X] ask `@publiclab/is-reviewers` for help, in a comment below
* [X] Clear-step functionality is working correct as expected.
> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software
Please make sure to get at least two reviews before asking for merging the PR as that would make the PR more reliable on our part
Thanks!

### CODE REVIEW

**Comment by ataata107** [comment]
I have written the code but I am not able to figure out its execution on my pc. I am running npm test-ui-2 but there are no logs.

**Comment by HarshKhandeparkar** [comment]
Does the process end or do you have to press `Ctrl + C` to kill it?

**Comment by ataata107** [comment]
@HarshKhandeparkar 
Ok apparantely I was able to get the result of `npm run test-ui-2` Mybe it was error of node version as I changed my node version to 8.17.0
As for verifying the clear step I have written the following code
```
test('Clear Step is pressed', async () => {
   page.on('dialog', async dialog => {
   await dialog.accept();
   });
   const valo = await page.evaluate(() => { document.querySelectorAll('#resetButton')[0].click(); });
   await page.waitForSelector('.step');  
}, timeout);
```
I am getting the following error
Screenshot from 2019-12-24 04-13-14

**Comment by ataata107** [comment]
For the following code
```
const timeout = process.env.SLOWMO ? 30000 : 10000;

const fs = require('fs');
beforeAll(async () => {
  console.log(page.url())
  path = fs.realpathSync('file://../examples/index.html');
  await page.goto('file://' + path, {waitUntil: 'domcontentloaded'});
  // await page.setCacheEnabled(true);
  console.log(page.url())
});

describe('Clear Steps', () => {
  test('Resize Module is added', async () => {
    
    await page.waitForSelector('.step');
    
    const Length = await page.evaluate(() => document.querySelectorAll('.step').length);
    await page.click('[data-value=\'resize\']');
    const Length1 = await page.evaluate(() => document.querySelectorAll('.step').length);
    expect(Length).toBe(1);
    expect(Length1).toBe(2);
  }, timeout);
  test('Brightness Module is added', async () => {
    await page.click('[data-value=\'brightness\']');
    const Length2 = await page.evaluate(() => document.querySelectorAll('.step').length);
    expect(Length2).toBe(3);
  }, timeout);
  test('Contrast Module is added', async () => {
    await page.click('[data-value=\'contrast\']');
    const Length3 = await page.evaluate(() => document.querySelectorAll('.step').length);
    expect(Length3).toBe(4);
  }, timeout);
  test('Saturation Module is added', async () => {
    await page.click('[data-value=\'saturation\']');
    const Length4 = await page.evaluate(() => document.querySelectorAll('.step').length);
    expect(Length4).toBe(5);
  }, timeout);
  test('Rotate Module is added', async () => {
    await page.click('[data-value=\'rotate\']');
    const Length5 = await page.evaluate(() => document.querySelectorAll('.step').length);
    expect(Length5).toBe(6);
  }, timeout);
  test('Crop Module is added', async () => {
    await page.click('[data-value=\'crop\']');
    const Length6 = await page.evaluate(() => document.querySelectorAll('.step').length);
    expect(Length6).toBe(7);
  }, timeout);
  test('Clear Step is pressed', async () => {
    
    
    page.on('dialog', async dialog => {
      console.log(dialog.type());
      (await dialog.accept());
      
    });
    
    // await page.setRequestInterception(true);
    //     page.on('request', interceptedRequest => {
    //         console.log('GOT NEW REQUEST', interceptedRequest.url());
    //         interceptedRequest.continue();
    //     });
    await page.click('#resetButton');
    try {
      // await page.goto('file:///home/ataata107/Desktop/dev/image-sequencer/examples/index.html',{waitUntil: 'domcontentloaded'})
      // let someVar = await page.waitForSelector('.step');
      console.log(page.url())
      // Rest of the code
      
      }
    catch (error) {
      console.log(error)
      }
    
    
  }, timeout);
  
});
    
    

```
    
The error I get


```
ataata107@ataata107-Inspiron-5559:~/Desktop/dev/image-sequencer$ node ./node_modules/.bin/jest
  console.log test/ui-2/test/clear_step.test.js:5
    about:blank

 PASS  test/ui-2/test/title.test.js
  Title of the page
    ✓ Title of the page (44ms)

  console.log test/ui-2/test/clear_step.test.js:9
    file:///home/ataata107/Desktop/dev/image-sequencer/examples/index.html

 PASS  test/ui-2/test/radio-group.test.js (14.978s)
  Radio Group
    ✓ Resize Module is added (4352ms)
    ✓ Brightness Module is added (1912ms)
    ✓ Contrast Module is added (2736ms)
    ✓ Saturation Module is added (1383ms)
    ✓ Rotate Module is added (796ms)
    ✓ Crop Module is added (967ms)

 FAIL  test/ui-2/test/clear_step.test.js (15.715s)
  Clear Steps
    ✓ Resize Module is added (5032ms)
    ✓ Brightness Module is added (1086ms)
    ✓ Contrast Module is added (3245ms)
    ✓ Saturation Module is added (1425ms)
    ✓ Rotate Module is added (998ms)
    ✓ Crop Module is added (649ms)
    ✕ Clear Step is pressed (391ms)

  ● Clear Steps › Clear Step is pressed

    ReferenceError: start is not defined

      at file::1:9
      at Suite.Object.<anonymous>.describe (test/ui-2/test/clear_step.test.js:48:3)

  ● Clear Steps › Clear Step is pressed

    ReferenceError: addRow is not defined

      at file::2:9
      at Suite.Object.<anonymous>.describe (test/ui-2/test/clear_step.test.js:48:3)

  ● Clear Steps › Clear Step is pressed

    ReferenceError: addRow is not defined

      at file::3:9
      at Suite.Object.<anonymous>.describe (test/ui-2/test/clear_step.test.js:48:3)

  ● Clear Steps › Clear Step is pressed

    ReferenceError: addRow is not defined

      at file::4:9
      at Suite.Object.<anonymous>.describe (test/ui-2/test/clear_step.test.js:48:3)

Test Suites: 1 failed, 2 passed, 3 total
Tests:       1 failed, 13 passed, 14 total
Snapshots:   0 total
Time:        17.066s
Ran all test suites.
  console.log test/ui-2/test/clear_step.test.js:52
    confirm

  console.log test/ui-2/test/clear_step.test.js:66
    file:///

```


Shouldn't pressing the clear button redirect to 

> file:///home/ataata107/Desktop/dev/image-sequencer/examples/index.html

But it redirects to 

> file:///

**Comment by HarshKhandeparkar** [comment]
Ah ok. Look at this code
https://github.com/publiclab/image-sequencer/blob/c370a8b4ea20267462f8fa559608618ca168e457/examples/demo.js#L85-L89

It redirects to `/` which would load the main page if we were using a web(HTTP) URL but we are using `file` protocol so it redirects to `/` which is your computer root. I think you should open a separate issue for this.

Or I'll do it. I have a fix in mind.

**Comment by ataata107** [comment]
Sure

**Comment by HarshKhandeparkar** [comment]
https://github.com/publiclab/image-sequencer/issues/1394

**Comment by ataata107** [comment]
@HarshKhandeparkar 
After modifying the changes as recommended in #1394 the clear step test is also working :). I will commit the changes once #1394 is resolved.

**Comment by HarshKhandeparkar** [comment]
👍

**Comment by ataata107** [comment]
Delete step is added, will uncomment clear step after #1399 is merged

File: test/ui-2/test/clear_step.test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  path = fs.realpathSync('file://../examples/index.html');
+  await page.goto('file://' + path, {waitUntil: 'domcontentloaded'});
+});
+
+describe('Clear Steps', () => {
+  test('Resize Module is added', async () => {
+    
+    await page.waitForSelector('.step');
+    
+    const Length = await page.evaluate(() => document.querySelectorAll('.step').length);
```
Comment (HarshKhandeparkar) [inline test/ui-2/test/clear_step.test.js]: Could you please add some line breaks in the below code. It is really difficult to read.

**Comment by jywarren** [comment]
Hi, all, how are we doing here? Thank you, this is looking great!!

**Comment by ataata107** [comment]
Hey, I guess this is ready to be merged.

**Comment by jywarren** [comment]
Thanks! Awesome :-)))) 🎉

**Pull request merged**